services/gitlab/gitlab-base.js

```diff
@@ -16,4 +16,50 @@ export default class GitLabBase extends BaseJsonService {
       })
     )
   }
+
+  async fetchPage({ page, requestParams, schema }) {
+    const { res, buffer } = await this._request({
+      ...requestParams,
+      ...{ options: { qs: { page } } },
+    })
+
+    const json = this._parseJson(buffer)
+    const data = this.constructor._validate(json, schema)
+    return { res, data }
+  }
+
+  async fetchPaginatedArrayData({
+    url,
+    options,
+    schema,
+    errorMessages,
+    firstPageOnly = false,
+  }) {
+    const requestParams = this.authHelper.withBasicAuth({
+      url,
+      options: {
+        headers: { Accept: 'application/json' },
+        qs: { per_page: 100 },
+        ...options,
+      },
+      errorMessages,
+    })
+
+    const {
+      res: { headers },
+      data,
+    } = await this.fetchPage({ page: 1, requestParams, schema })
+    const numberOfPages = headers['x-total-pages']
+
+    if (numberOfPages === 1 || firstPageOnly) {
+      return data
+    }
+
+    const pageData = await Promise.all(
+      [...Array(numberOfPages - 1).keys()].map((_, i) =>
+        this.fetchPage({ page: ++i + 1, requestParams, schema })
+      )
+    )
+    return [...data].concat(...pageData)
+  }
```

Comment:
GitLab's max per page value is 100, and I saw that exceeded in several cases just with the releases on the handful of projects I tested with. A cursory glance through the GitLab API docs makes it seem a very common response structure will be pages of arrays with the page keys in the response headers.

Opted to go ahead and pull that into the base class in anticipation of other APIs. Added it with relevant short circuiting logic (since GitLab doesn't seem to have a separate "latest" API endpoint that returns a single object like GitHub does in some cases), and also requesting all the pages in parallel since it's both possible to do so and preferable to 1 by 1